drivers/adc/ad7606/ad7606.c

```diff
@@ -965,16 +1049,18 @@ static int32_t ad7606_read_one_sample(struct ad7606_dev *dev, uint32_t * data)
  *
  * @param dev        - The device structure.
  * @param data       - Pointer to location of buffer where to store the data.
+ * @param stats      - Pointer to location of buffer where to store the status info.
  * @param samples    - Number of samples to read
  *
  * @return ret - return code.
  *         Example: -EIO - SPI communication error.
  *                  -ETIME - Timeout while waiting for the BUSY signal.
  *                  -EBADMSG - CRC computation mismatch.
+ * 					-ENOMEM  - No mem pointer for stutas
  *                  0 - No errors encountered.
 *******************************************************************************/
 int32_t ad7606_read_samples(struct ad7606_dev *dev, uint32_t * data,
-			    uint32_t samples)
+			    uint32_t samples,uint8_t * stats)
```

Comment:
You're changing the function parameter list, so existing code will break. It would be better if the status info may be read using a new function instead.

Reply:
thx for reply. 

It was my intention to modify the ad7606_read_samples function, because I feel it's a bit redundant that a new function to be made for split status info and adc data, for this way user needs to call 2 functions( and prepare 3 arrays maybe, one for mixed data, one for signed data and one for info) just for one purpose. 

So is making a split function the only way?